src/emacs/lean-server.el

```diff
@@ -44,8 +47,9 @@
        (setf (lean-server-session-tasks sess) res)
        (lean-server-notify-tasks-changed sess old-tasks)))
     ("error"
-     (message "error: %s" (plist-get res :message))
-     ; TODO(gabriel): maybe even add the error as a message
+     (when lean-server-print-errors-p
+       (message "error: %s" (plist-get res :message)))
```

Comment:
This should only happen with protocol errors, i.e., malformed messages or unknown commands.  I don't think we should hide them.  Did you encounter this kind of error in a different situation?